lib/optimizer/src/Transformer/AmpBoilerplate.php

```diff
@@ -74,9 +75,10 @@ public function transform(Document $document, ErrorCollection $errors)
     private function removeStyleAndNoscriptTags(Document $document)
     {
         $headNode = $document->head->firstChild;
+        $devMode = DevMode::isActiveForDocument($document);
         while ($headNode) {
             $nextSibling = $headNode->nextSibling;
-            if ($headNode instanceof DOMElement) {
+            if ($headNode instanceof DOMElement && (! $devMode || ! DevMode::hasExemptionForNode($headNode))) {
```

Comment:
Why not use `isExemptFromValidation`?

```suggestion
            if ($headNode instanceof DOMElement && ! DevMode::isExemptFromValidation($headNode)) {
```

Then `$devMode` variable could be removed.

Reply:
Because it's a loop, and that would do the check against the document multiple times although it's only needed once.

Reply:
OK. But that's almost always true for other instances of where `isExemptFromValidation` is used currently. 

I don't feel strongly about this.